docs/guides/apphub-setup-changelog.md

```diff
@@ -0,0 +1,197 @@
+
+## What are changelogs
+
+A changelog is a file which contains a curated, chronologically ordered list of notable changes for each version of a project [https://keepachangelog.com/en/1.1.0/#what](https://keepachangelog.com/en/1.1.0/#what).
+
+A changelog is typically saved as `CHANGELOG.md` file in the root of the project. This is [an example](https://github.com/dhis2/route-manager-app/blob/main/CHANGELOG.md) of such a file.
+
+
+### Semantic Versioning
+
+While there are a variety of ways that you can create and maintain application versions, [semver](https://semver.org/) specification provides the most popular versioning scheme and is used widely in open-source projects including DHIS2. Going through the details of semver is beyond the scope of this guide but there are plenty of resources online describing the scheme and its details - This [article from Baeldung](https://www.baeldung.com/cs/semantic-versioning) provides a good introduction.
+
+
+### Conventional commits
+
+`Semver` tackles the versioning side of the task, but this does not answer how to keep a changelog with the changes in each version.
+
+In order to make the process of creating changelogs reliable and automated, the most common approach is to make it driven by the commit messages. In order to do so, the commit messages should follow a standard format. A very popular such format is the [conventional commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary).
+
+The basic idea is to write a commit message using the format `<type>[optional scope]: <description>`, for example: `feat(ui): update the header design`. 
+
+This can be picked up by an automated tool then (such as `semantic-release` described shortly) and decide the next version number based on the commit type (i.e. `feat` in the previous example) and update the changelog automatically with the commit message.
+
+> This makes it crucial that developers are disciplined in writing _good_ commit messages. We provide several resources at the end of the article about writing good messages, but the crucial aspect - as a developer - is to realise that your commit messages will be read by other developers and end-users (through AppHub in this specific context) so they clearly communicate what changed in a commit in a way understandable for all these stake holders.
+
+## How are changelog picked by App Management
+
+There are few conditions necessary in order for App Hub and App Management app to pick up and display the changes:
+
+1. The changelog should be saved in `CHANGELOG.md` file in the root of the project.
+
+1. The changelog should appear in `master` or `main` branch.
+
+This might be limiting for some setups that do not use these conventions, but we believe this covers the majority of use cases and in the future, we will look into making these requirements configurable if there is a need.
```

Comment:
```suggestion
This might be limiting for some setups that do not use these conventions, but we believe this covers the majority of use cases, and in the future, we will look into making these requirements configurable if there is a need.
```